cdap-api-spark/src/main/scala/co/cask/cdap/api/spark/SparkMain.scala

```diff
@@ -133,6 +133,70 @@ trait SparkMain extends Serializable {
       * Creates a [[org.apache.spark.rdd.RDD]] from the given [[co.cask.cdap.api.dataset.Dataset]].
       *
       * @param datasetName name of the Dataset
+      * @tparam K key type
+      * @tparam V value type
+      * @return A new [[org.apache.spark.rdd.RDD]] instance that reads from the given Dataset.
+      * @throws co.cask.cdap.api.data.DatasetInstantiationException if the Dataset doesn't exist
+      */
+    def fromDataset[K: ClassTag, V: ClassTag](datasetName: String)
```

Comment:
Discussed in person.

This is cause overloading a function with default parameters in scala will raise ambiguity. As a result, we have to  expand all these functions.